editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -97,219 +110,567 @@ void VersionControlEditorPlugin::popup_vcs_set_up_dialog(const Control *p_gui_ba
 	vcs_interface->set_script_and_instance(script.get_ref_ptr(), addon_script_instance);
 
 	EditorVCSInterface::set_singleton(vcs_interface);
+
+	return true;
+}
+
+void VersionControlEditorPlugin::_set_up() {
+	register_editor();
 	EditorFileSystem::get_singleton()->connect("filesystem_changed", this, "_refresh_stage_area");
 
 	String res_dir = OS::get_singleton()->get_resource_dir();
 
 	ERR_FAIL_COND_MSG(!EditorVCSInterface::get_singleton()->initialize(res_dir), "VCS was not initialized");
 
 	_refresh_stage_area();
+	_refresh_commit_list();
+	_refresh_branch_list();
 }
 
-void VersionControlEditorPlugin::_send_commit_msg() {
-	if (EditorVCSInterface::get_singleton()) {
-		if (staged_files_count == 0) {
-			commit_status->set_text(TTR("No files added to stage"));
-			return;
-		}
+void VersionControlEditorPlugin::_set_credentials() {
+	if (set_up_username->get_text() == "" || set_up_password->get_text() == "") {
+		WARN_PRINT("Some features will not work without remote credentials.")
+	}
 
-		EditorVCSInterface::get_singleton()->commit(commit_message->get_text());
+	EditorVCSInterface::get_singleton()->set_up_credentials(set_up_username->get_text(), set_up_password->get_text());
+	set_up_username->clear();
+	set_up_password->clear();
+}
 
-		commit_message->set_text("");
-		version_control_dock_button->set_pressed(false);
-	} else {
-		WARN_PRINT("No VCS addon is initialized. Select a Version Control Addon from Project menu");
+void VersionControlEditorPlugin::_refresh_branch_list() {
+	ERR_FAIL_COND_MSG(!EditorVCSInterface::get_singleton(), "No VCS plugin is initialized. Select a Version Control Addon from Project menu");
+
+	List<String> branch_list = EditorVCSInterface::get_singleton()->get_branch_list();
+	branch_select->clear();
+	for (int i = 0; i < branch_list.size(); i++) {
+		branch_select->add_item(branch_list[i], i);
+	}
+
+	// TODO: Add new brach feature
+	// branch_select->add_separator();
+	// branch_select->add_item("New Branch");
+	branch_select->select(0); // First branch in the list is current branch
+}
+
+void VersionControlEditorPlugin::_refresh_commit_list() {
+	ERR_FAIL_COND_MSG(!EditorVCSInterface::get_singleton(), "No VCS plugin is initialized. Select a Version Control Addon from Project menu");
+
+	commit_list->get_root()->clear_children();
+
+	List<EditorVCSInterface::Commit> commit_info_list = EditorVCSInterface::get_singleton()->get_previous_commits();
+
+	for (List<EditorVCSInterface::Commit>::Element *E = commit_info_list.front(); E; E = E->next()) {
+		EditorVCSInterface::Commit commit = E->get();
+		TreeItem *item = commit_list->create_item(commit_list->get_root());
+		item->set_text(0, commit.msg.strip_edges());
+		item->set_text(1, commit.author.strip_edges());
+		item->set_metadata(0, commit.hex_id);
 	}
+}
+
+void VersionControlEditorPlugin::_commit() {
+	ERR_FAIL_COND_MSG(!EditorVCSInterface::get_singleton(), "No VCS plugin is initialized. Select a Version Control Addon from Project menu");
```

Comment:
There are multiple calls with the same string so you can probably create a function that prints this and call it whenever needed. Probably named `_ensure_vcs_plugin_is_initialized()` or something similar